NEW CMS 5 compatability (#41)
silverstripe/silverstripe-module: README.md

> # Silverstripe supported module skeleton

Reviewer: ```suggestion
# Silverstripe CMS supported module skeleton
```

Reviewer 2: Done

> # Silverstripe supported module skeleton

Reviewer: ```suggestion
# Silverstripe CMS supported module skeleton
```

Reviewer 2: Isn't this the same as https://github.com/silverstripe/silverstripe-module/pull/41#discussion_r1189203791?

placeholders are edited where relevant
- Review the README files in the various provided directories. You should ultimately delete these README files when you have added your code
- Update the module's `composer.json` with your requirements and package name
- Update (or remove) `package.json` with your requirements and package name. Run `yarn install` (or remove `yarn.lock`) to
ensure dependencies resolve correctly
- Clear the git history by running `rm -rf .git && git init`
- Add and push to a VCS repository
- Either [publish](https://getcomposer.org/doc/02-libraries.md#publishing-to-packagist) the module on packagist.org, or add a [custom repository](https://getcomposer.org/doc/02-libraries.md#publishing-to-a-vcs) to your main `composer.json`
- Require the module in your main `composer.json`
- If you need to build your css or js and are using components, injector, scss variables, etc from `silverstripe/admin`:
  - Ensure that `silverstripe/admin` is installed with `composer install --prefer-source` instead of the default `--prefer-dist` (you can use `composer reinstall silverstripe/admin --prefer-source` if you already installed it)

>   - If you are relying on additional dependencies from `silverstripe/admin` instead of adding them as dependencies in your `package.json` file, you need to install third party dependencies in `silverstripe/admin` by running `yarn install` in the `vendor/silverstripe/admin/` directory.

Reviewer: We need a better way of explaining this thing. But that will do for now.

- Start developing your module!

## License

See [License](LICENSE.md)

We have included a 3-clause BSD license you can use as a default. We advocate for the BSD license as

> it is one of the most permissive and open licenses.

Reviewer: I'm not sure we "advocate" a specific license.

```suggestion
This module template defaults to using the "BSD-3-Clause" license. The BSD-3 license is one of the most
permissive open-source license and is used by most Silverstripe CMS module.
```

Reviewer: Just realised this was copied from somewhere else. If my suggesting wording is controversial, I'm happy to ditch it.

Reviewer 2: Done

Feel free to alter the [license.md](LICENSE.md) to suit if you want to use an alternative license. Make sure

> you also update the license in your `composer.json` file if you do change the license for your project.

Reviewer: I think this is a bit more to the point.
```suggestion
To publish your module under a different license:
- update the [`license.md`](LICENSE.md) file
- update the `license' key in your [`composer.json`](composer.json).
```

Reviewer 2: Done

> You can use [choosealicense.com](https://choosealicense.com) to help pick a suitable license for your project.

Reviewer: ```suggestion
You can use [choosealicense.com](https://choosealicense.com) to help you pick a suitable license for your project.
```

Reviewer 2: Done
